### PR TITLE
KO-13263 Added module exports to support UMD

### PR DESCRIPTION
**Pull request by abdulllaaahhh**
This change is [Reviewable](https://reviewable.io/reviews/kounta/spi-client-js/1)

### CODE REVIEW

**Review by brendan-roche — CHANGES_REQUESTED** [review]
I'd suggest squashing your changes into a single commit, and then submit a PR to main library

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/kounta/spi-client-js/1#-:-LP8Bjre2h3MUhTGg6kx:b-puic1u)* status: all files reviewed, 7 unresolved discussions (waiting on @muhammadkounta)

___
*[package.json, line 4 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP88Luf4qnG5irGKBvN:-LP88LugC05SeAim0s_m:bg2py65) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/package.json#L4)):*
> ```JSON
>   "name": "@assemblypayments/kounta-spi-client-js",
>   "version": "1.0.0",
>   "description": "Modified JavaScript Client Library for Assembly Payments' In-Store Integration.",
> ```

We shouldn't be altering the name, version or description of another library. Ideally we've just forked it temporarily until they update and we don't need to
___
*[package.json, line 7 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89NVe5DB_YmDZb1_W:-LP89NVe5DB_YmDZb1_X:b-g04cfb) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/package.json#L7)):*
> ```JSON
>   "main": "karma.conf.js",
>   "scripts": {
>     "test": "echo 1",
> ```

why was this changed?
___
*[package.json, line 34 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89T6UA96c9OZsPYTc:-LP89T6UA96c9OZsPYTd:bs9ecnf) ([raw file](https://github.com/kounta/spi-client-js/blob/dc2e62ead35edbb97094a06fccaa98a92701d41a/package.json#L34)):*
> ```JSON
>   },
>   "bugs": {
>     "url": "https://github.com/AssemblyPayments/spi-client-js/issues"
> ```

we're not really maintaining ourselves so it should still point to original library issues
___
*[dist/spi-client-js.js, line 291 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89lHA-LrOYOZr4eOa:-LP89lHA-LrOYOZr4eOb:blxcd1c) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/dist/spi-client-js.js#L291)):*
> ```JavaScript
> var _aesJs = require('aes-js');
> 
> var _aesJs2 = _interopRequireDefault(_aesJs);
> ```

why are they prefixed with underscores?
___
*[dist/spi-client-js.js, line 396 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP8AClH5oD_-_VQIFwY:-LP8AClH5oD_-_VQIFwZ:b-3tiitn) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/dist/spi-client-js.js#L396)):*
> ```JavaScript
> var _jssha = require('jssha');
> 
> var _jssha2 = _interopRequireDefault(_jssha);
> ```

try to keep all requires together, and ideally at the top of the file
___
*[dist/spi-client-js.js, line 398 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP8A8ONDo1JDAc-ZTb2:-LP8A8ONDo1JDAc-ZTb3:b-5fksa3) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/dist/spi-client-js.js#L398)):*
> ```JavaScript
> var _jssha2 = _interopRequireDefault(_jssha);
> 
> function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
> ```

You've got this defined at this top of this file too?
___
*[src/Crypto.js, line 2 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP8AdKu1qEUSuoIJIem:-LP8AdKu1qEUSuoIJIen:b-3b0nya) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/src/Crypto.js#L2)):*
> ```JavaScript
> import jsSHA from 'jssha';
> import aesjs from 'aes-js';
> ```

how come sometimes you're using common js (require) and other times es6 imports?

**Comment by abdulllaaahhh** [comment]
___
*[src/Crypto.js, line 2 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP8AdKu1qEUSuoIJIem:-LP8FinR88VNyPvSp0xL:b-dmoc9d) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/src/Crypto.js#L2)):*
<details><summary><i>Previously, brendan-roche (Brendan Roche) wrote…</i></summary><blockquote>

how come sometimes you're using common js (require) and other times es6 imports?
</blockquote></details>

Basically i . only used the es6 imports, the dist/spi-client-js file is the build file generated through "npm run build"

**Comment by abdulllaaahhh** [comment]
___
*[package.json, line 7 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89NVe5DB_YmDZb1_W:-LP8HQ-M3cYAioe86X6B:bdxrkvl) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/package.json#L7)):*
<details><summary><i>Previously, brendan-roche (Brendan Roche) wrote…</i></summary><blockquote>

why was this changed?
</blockquote></details>

their karma test was returning errors on npm test and as we have just forked it temporarily so the test is by-passed

**Comment by abdulllaaahhh** [comment]
___
*[package.json, line 34 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89T6UA96c9OZsPYTc:-LP8Jhk22P7fSAKey6cd:bx5iw9k) ([raw file](https://github.com/kounta/spi-client-js/blob/dc2e62ead35edbb97094a06fccaa98a92701d41a/package.json#L34)):*
<details><summary><i>Previously, brendan-roche (Brendan Roche) wrote…</i></summary><blockquote>

we're not really maintaining ourselves so it should still point to original library issues
</blockquote></details>

its in devDependencies so it won't install on npm install

**Comment by abdulllaaahhh** [comment]
___
*[package.json, line 4 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP88Luf4qnG5irGKBvN:-LP8KPDCEaPGDdgQsIQD:b9uetde) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/package.json#L4)):*
<details><summary><i>Previously, brendan-roche (Brendan Roche) wrote…</i></summary><blockquote>

We shouldn't be altering the name, version or description of another library. Ideally we've just forked it temporarily until they update and we don't need to
</blockquote></details>

changed this incase we use the npm install because if the name is same and we run npm install <package_name> it will point to the main package not the forked one,

**Comment by abdulllaaahhh** [comment]
___
*[dist/spi-client-js.js, line 291 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89lHA-LrOYOZr4eOa:-LP8LyaUALORzxtNyTlG:b-qqvrqw) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/dist/spi-client-js.js#L291)):*
<details><summary><i>Previously, brendan-roche (Brendan Roche) wrote…</i></summary><blockquote>

why are they prefixed with underscores?
</blockquote></details>

its a build file modified through npm run build

**Comment by abdulllaaahhh** [comment]
___
*[dist/spi-client-js.js, line 396 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP8AClH5oD_-_VQIFwY:-LP8M3UD8Dvjf34JNITk:b-qqvrqw) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/dist/spi-client-js.js#L396)):*
<details><summary><i>Previously, brendan-roche (Brendan Roche) wrote…</i></summary><blockquote>

try to keep all requires together, and ideally at the top of the file
</blockquote></details>

its a build file modified through npm run build

**Comment by abdulllaaahhh** [comment]
___
*[dist/spi-client-js.js, line 398 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP8A8ONDo1JDAc-ZTb2:-LP8M4nM91puru1tszbw:b-qqvrqw) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/dist/spi-client-js.js#L398)):*
<details><summary><i>Previously, brendan-roche (Brendan Roche) wrote…</i></summary><blockquote>

You've got this defined at this top of this file too?
</blockquote></details>

its a build file modified through npm run build

**Review by brendan-roche — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/kounta/spi-client-js/1)* status: all files reviewed, 7 unresolved discussions (waiting on @brendan-roche)

___
*[package.json, line 7 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89NVe5DB_YmDZb1_W:-LP8LBgFAWgmlPX9ocl3:bqniqw0) ([raw file](https://github.com/kounta/spi-client-js/blob/4c62596dee2b7797a95bc441f331f9eaf786d959/package.json#L7)):*
<details><summary><i>Previously, muhammadkounta (Muhammad) wrote…</i></summary><blockquote>

their karma test was returning errors on npm test and as we have just forked it temporarily so the test is by-passed
</blockquote></details>

It's our responsibility to fix those test only if the changes we've made break it. Otherwise it shouldn't concern us.
___
*[package.json, line 34 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89T6UA96c9OZsPYTc:-LP8Lyt004tybhl9AEb2:b-l4c4wv) ([raw file](https://github.com/kounta/spi-client-js/blob/dc2e62ead35edbb97094a06fccaa98a92701d41a/package.json#L34)):*
<details><summary><i>Previously, muhammadkounta (Muhammad) wrote…</i></summary><blockquote>

its in devDependencies so it won't install on npm install
</blockquote></details>

for some reason reviewable has moved my comment under a different line. If you change the review for this file to include all changes, you'll see my comments relate to removing:
```
  "bugs": {
    "url": "https://github.com/AssemblyPayments/spi-client-js/issues"
```

**Comment by abdulllaaahhh** [comment]
___
*[package.json, line 34 at r1](https://reviewable.io/reviews/kounta/spi-client-js/1#-LP89T6UA96c9OZsPYTc:-LP8Vy2r7za_M5Nyb628:b4vx31n) ([raw file](https://github.com/kounta/spi-client-js/blob/dc2e62ead35edbb97094a06fccaa98a92701d41a/package.json#L34)):*
<details><summary><i>Previously, brendan-roche (Brendan Roche) wrote…</i></summary><blockquote>

for some reason reviewable has moved my comment under a different line. If you change the review for this file to include all changes, you'll see my comments relate to removing:
```
  "bugs": {
    "url": "https://github.com/AssemblyPayments/spi-client-js/issues"
```
</blockquote></details>

got it, removed it as its not affecting anything

**Review by brendan-roche — COMMENTED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/kounta/spi-client-js/1)* status: all files reviewed, 1 unresolved discussion (waiting on @muhammadkounta)
